src/backend/utils/request-parser.ts

```diff
@@ -14,27 +15,29 @@ import BaseResource from '../adapters/base-resource'
 const RequestParser = (originalRequest: ActionRequest, resource: BaseResource): ActionRequest => {
   const { payload: originalPayload } = originalRequest
 
-  const payload = Object.entries(originalPayload || {}).reduce((memo, [path, value]) => {
+  const payload = Object.entries(originalPayload || {}).reduce((memo, [path, formValue]) => {
     const property = resource.property(path)
```

Comment:
here is a known issue - `resource.property` doesn't take into consideration nested properties and options. So when user changes type of the property it wont be affected here. The same goes for nested properties - they wont be properly converted back.

I am on it